src/Uno.Extensions.Reactive.Generator/Rules.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Uno.Extensions.Reactive.Generator;
+
+[SuppressMessage("ReSharper", "InconsistentNaming")]
+internal static class Rules
+{
+	// references:
+	//	Categories: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/categories
+
+	// General usage [0000-0999]
+
+	// Bindings [1000-1999]
+
+	// Commands [2000-2999]
+	public static class FEED2001
+	{
+		private const string message = "There is no public property '{0}' on the class '{1}' that can be used as command parameter for '{2}'";
+
+		public static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
+			nameof(FEED2001),
+			"Invalid feed property name",
+			message,
+			Category.Usage,
+			DiagnosticSeverity.Error,
+			isEnabledByDefault: true);
+
+		public static string GetMessage(INamedTypeSymbol @class, IMethodSymbol method, string missingPropertyName)
+			=> string.Format(CultureInfo.InvariantCulture, message, missingPropertyName, method.Name, @class.Name);
+
+		public static Diagnostic GetDiagnostic(INamedTypeSymbol @class, IMethodSymbol method, IParameterSymbol parameter)
+			=> Diagnostic.Create(
+				Descriptor,
+				parameter.DeclaringSyntaxReferences.FirstOrDefault() is { } syntax
+					? Location.Create(syntax.SyntaxTree, syntax.Span)
+					: Location.None,
+				parameter.Name,
+				@class.Name,
+				method.Name);
+	}
+
+	public static class FEED2002
+	{
+		private const string message = "The property '{0}' resolved on the class '{1}' cannot be used as command parameter for '{2}' as it's not a Feed";
+
+		public static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
+			nameof(FEED2002),
+			"Invalid property type",
```

Comment:
Invalid what way ? What can be done?

Reply:
This is only the "title", the "message" already gives more details, and I'm adding a direct link to the doc